web/src/inbox_ui.js

```diff
@@ -71,6 +71,12 @@ function should_include_muted() {
 }
 
 export function show() {
+    // Avoid setting col_focus to recipient when moving to inbox from other narrows.
+    // We prefer to focus entire row instead of stream name for inbox-header.
+    if (col_focus === COLUMNS.RECIPIENT) {
+        col_focus = COLUMNS.COLLAPSE_BUTTON;
+    }
```

Comment:
Hmm, where does this logic limit this behavior to the inbox-header rows? Is there no `RECIPIENT` column in the normal rows? Might want to make that more obvious at this call point.

Reply:
Added a comment:
```
    // Since inbox-row doesn't has a collapse button, focus on COLUMNS.COLLAPSE_BUTTON
    // is same as focus on COLUMNS.RECIPIENT. See `set_list_focus` for details.
```